netops_demo/py/functions/generate/generate.py

```diff
@@ -105,6 +105,18 @@ def _sites(context):
     return sites
 
 
+def _devices(context):
+    deployment = context.user_data.deployment
+    context.logger.info_with('Sending list of devices', deployment=deployment)
+
+    devices = []
+    for company in deployment.companies:
+        for i, site_locations in company.components.items():
+            for j, x in enumerate(site_locations['devices']):
```

Comment:
Please don't use `j`, `x`. I'm also not a fan of `i` - there's no good reason not to name variables as what they are so that a person doesn't need to constantly check what they contain. 
```
i = site_location_index
j = device_index
x = device_id (i think)
```